luigi/scheduler.py

```diff
@@ -234,6 +234,27 @@ def load(self):
         else:
             logger.info("No prior state file exists at %s. Starting with clean slate", self._state_path)
 
+    def cascade_disabled_state(self, tasks):
+
+        """ Iterate through all pending tasks, setting the status to DISABLED
+        if at least one of it's dependencies is DISABLED and the rest are
+        either DONE or DISABLED. """
+
+        for task in tasks:
+            if task.status != PENDING:
+                continue
+
+            any_disabled = False
+            should_disable = True
+            for dep in task.deps:
+                t = self.get_task(dep)
+                if t.status == DISABLED:
+                    any_disabled = True
+                should_disable &= (t.status == DISABLED or t.status == DONE)
```

Comment:
i guess i'm confused what's the difference between `any_disabled` and `should_disable`